core/drivers/crypto/crypto_api/acipher/ecc.c

```diff
@@ -340,10 +341,145 @@ static TEE_Result ecc_shared_secret(struct ecc_keypair *private_key,
 	return ret;
 }
 
+static TEE_Result ecc_encrypt(struct ecc_public_key *key,
+			      const uint8_t *src, size_t src_len,
+			      uint8_t *dst, size_t *dst_len)
+{
+	TEE_Result ret = TEE_ERROR_BAD_PARAMETERS;
+	struct drvcrypt_cipher_data cdata = { };
+	struct drvcrypt_ecc *ecc = NULL;
+	size_t output_size = 0;
+	size_t size_bytes = 0;
+
+	if (!key || !src || !dst) {
+		CRYPTO_TRACE("Input parameters reference error");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	ecc = drvcrypt_get_ops(CRYPTO_ECC);
+	if (!ecc || !ecc->encrypt)
+		return TEE_ERROR_NOT_IMPLEMENTED;
+
+	if (key->curve != TEE_ECC_CURVE_SM2)
+		return TEE_ERROR_NOT_IMPLEMENTED;
+
+	size_bytes = get_ecc_key_size_bytes(key->curve);
+	if (!size_bytes) {
+		CRYPTO_TRACE("Invalid size_bytes");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	output_size = 2 * size_bytes + src_len + TEE_SM3_HASH_SIZE;
+
+	/* Verify the ciphertext length function of the key size */
```

Comment:
nitpick. I'd remove this comment and align the size above just like you did in decrypt. I think it is nicer that way